woocommerce/woocommerce-gateway-stripe
# Add support to UPE on E2E tests

Closes #2589 

## Changes proposed in this Pull Request:

- Add support to UPE on the E2E tests
- Add the Disable Emails plugin in the E2E setup
- Add support to multiple error messages in the card failures (on UPE the messages are a bit different).





## Testing instructions

1. Follow the [instructions](https://github.com/woocommerce/woocommerce-gateway-stripe/tree/develop/tests/e2e#environment-setup) to setup E2E tests in a JN site.
2. Run all the tests in the JN site created on step 1. All tests should pass.
3. Enable UPE in the JN site being tested.
4. Run all the tests in the JN site created on step 1. All tests should pass.

Note: during the tests, you shouldn't receive emails from the JN site thanks to the Disable emails plugin. Before the plugin you would receive about 20 emails from the test orders.

---


**Post merge**
